### PR TITLE
Delete "client library for java" suffix in all readme.md

### PR DESCRIPTION
**Pull request by chenrujun**
Delete "client library for java" suffix in all readme.md.
The suffix is required because the CI check in azure-sdk-for-java repo.
Now such kind of check does not exist anymore. So, it can be deleted.

### CODE REVIEW

File: aad/spring-cloud-azure-starter-active-directory/web-client-access-resource-server/aad-resource-server-obo/README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Azure OAuth 2.0 Sample for Azure AD Spring Boot Starter Resource Server Obo client library for Java
+# Azure OAuth 2.0 Sample for Azure AD Spring Boot Starter Resource Server Obo
```
Comment (saragluna) [inline aad/spring-cloud-azure-starter-active-directory/web-client-access-resource-server/aad-resource-server-obo/README.md:1]: Do you think `On-Behalf-Of flow` or `OBO` is better? 

https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Sample for Spring JMS with Azure Service Bus Queue Spring Cloud client library for Java 
+# Sample for Spring JMS with Azure Service Bus Queue Spring Cloud
```
Comment (saragluna) [inline servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md:1]: We can remove `Spring Cloud` now as well.

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-topic/README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Sample for Spring JMS with Azure Service Bus Topic Spring Cloud client library for Java 
+# Sample for Spring JMS with Azure Service Bus Topic Spring Cloud
```
Comment (saragluna) [inline servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-topic/README.md:1]: Same here.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: aad/spring-cloud-azure-starter-active-directory/web-client-access-resource-server/aad-resource-server-obo/README.md (hunk 1 above)
Comment (chenrujun) [reply aad/spring-cloud-azure-starter-active-directory/web-client-access-resource-server/aad-resource-server-obo/README.md:1]: There are much more improvements in all titles.  In this PR, I just **Delete "client library for java" suffix in all readme.md** as described in the title.

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md (hunk 2 above)
Comment (chenrujun) [reply servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-queue/README.md:1]: Same here.

File: servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-topic/README.md (hunk 3 above)
Comment (chenrujun) [reply servicebus/spring-cloud-azure-starter-servicebus-jms/servicebus-jms-topic/README.md:1]: Same here.

**Review by saragluna — APPROVED** [review]

**Pull request merged**